hooks/pre_validations/05-puppet_certs_exist.rb

```diff
@@ -0,0 +1,27 @@
+def error(message)
+  say message
+  logger.error message
+  kafo.class.exit 101
+end
+
+def param_value(mod, name)
+  param(mod, name).value if param(mod, name)
+end
+
+server_crl_path = param_value('foreman', 'server_ssl_crl')
+puppet_server_implementation = param_value('puppet', 'server_implementation')
+puppet_passenger = param_value('puppet', 'server_passenger')
+cert_dir = param_value('puppet', 'server_ssl_dir')
+puppet_ca_cert = File.join(cert_dir, 'ca/ca_crt.pem')
+puppet_enabled = kafo.module('puppet').enabled?
+
+client_message = "- is puppet already installed without puppet CA? You can remove the existing certificates with 'rm -rf #{cert_dir}' to get puppet CA properly configured."
+
+if cert_dir && File.directory?(cert_dir)
+  if !server_crl_path.empty? && !File.exists?(server_crl_path)
+    error "The file #{server_crl_path} does not exist.\n #{client_message}\n - if you set custom revocation list (--foreman-server-ssl-crl) make sure the file exists."
+  end
+  if puppet_enabled && puppet_passenger && puppet_server_implementation == 'master' && !File.exists?(puppet_ca_cert)
```

Comment:
@domcleal, from my experiments it seems that the default installation fails only when the apache starts the puppet master. I'm not sure if Foreman can be installed without Puppet CA or puppet master so I limited the CA cert check to just when the puppet master is used. Wasn't it valid concern?